Fix incremental update

julmud opened the pull request.

julmud committed:
Fix incremental updates for cast and crew

In case of an incremental update, the in memory table of cast
and crews wasn't correctly initialised. This led to duplicate errors
when running the SQL requests for inserting the new people.

julmud committed:
Fix a few more notices

Fix some invalid offset notices, which are quite frustrating when
debugging or your PHP installation is configured for displaying
notices.

julmud committed:
Fix creation of temporary tables for MySQL 5.7.28

The 'TYPE=MyISAM' option when creating a temporary table is invalid
since at least MySQL 4.1. With our use cases, it's better to let MySQL
decide by itself what type of temporary table it should use
(by default: in memory, and if the table gets too big MyISAM before 5.7,
InnoDB after that).

gjnoonan reviewed (changes requested):
I wish GitHub allowed you to comment on git commit messages. Could we have a bit more information on the last 2, please?

> Fix a few more notices

What generated these notices, were they causing an error?

> Fix creation of temporary tables for MySQL 5.7.28 

doesn't tell me what was broken, or why this fix is needed without having to look into it.

In combination with no description on the PR I am unsure what I am reviewing.

That being said, from a code point of view everything looks good to go, I have been unable to thoroughly test as the update has not been broken for me, however.

julmud commented:
> That being said, from a code point of view everything looks good to go, I have been unable to thoroughly test as the update has not been broken for me, however.

The problem happens for an _incremental_ update, i.e. when there's already a collection loaded a new collection file is uploaded with some new discs in it. In that case, the fast update fails twice (with a SQL error), and the importer then disables the fast update and eventually successfully updates the database. This fixes the fast update in that case.

julmud commented:
I've reworded the commit comments

julmud committed:
Remove duplicated code from summary.php

Instead of duplicating the code of some functions in the summary.php
code simply to support them in the included global.php or siteconfig.php
files, it's better to include functions.php itself, so that we're certain
there's no difference in behaviour.
